Foundation/NSTask.swift

```diff
@@ -194,11 +189,69 @@ public class NSTask : NSObject {
             argv.dealloc(args.count + 1)
         }
         
-        // Lock...
+        var taskSocketPair : [Int32] = [0, 0]
+        socketpair(AF_UNIX, SOCK_STREAM, 0, &taskSocketPair)
         
-        processLaunchLock.lock()
+        var context = CFSocketContext(version: 0, info: UnsafeMutablePointer<Void>(Unmanaged.passUnretained(self).toOpaque()),
+                                               retain: runLoopSourceRetain, release: runLoopSourceRelease, copyDescription: nil)
         
-        // ...and load
+        let socket = CFSocketCreateWithNative( nil, taskSocketPair[0], CFSocketCallBackType.DataCallBack.rawValue, {
```

Comment:
Got it, on the way.

Also, are we certain that inline closures for ```@convention(c)``` arguments are meant to be illegal?  The official docs state:

```C function pointers are imported into Swift as closures with C function pointer calling convention, denoted by the @convention(c) attribute.```

Even mikeash flagged it as one of the big Swift 2's big improvements, as a replacement for CFunctionPointer.  (https://mikeash.com/pyblog/friday-qa-2015-06-19-the-best-of-whats-new-in-swift.html)